### ISSUE TITLE
make GAF of GO annotations in contig files

### ISSUE THREAD

**Issue by mah11** [pombase/pombase-chado#861]
As described in email - create a GAF from the GO annotations in the *.contig files so we don't have to carry on editing in Artemis to use evidence not available in Canto. I'm rashly assuming it can go in pombe-embl/external_data/external-go-data/, to be included in the main GO file generation.

When it's done, ping me and I'll delete all the annotations from the contig files and we can enjoy the glorious GAF-based future.

**Comment by kimrutherford** [comment]
I've had an attempt at making a GAF file from the annotation in the contigs. It looks OK at first glance but I'm still checking.

[pombase-build-contig-files-only.gaf.txt](https://github.com/pombase/pombase-chado/files/7255200/pombase-build-contig-files-only.gaf.txt)

**Comment by kimrutherford** [comment]
I'm now doing a test load of the contigs into a fresh Chado database but with the GO loading disabled.  Then I'll load the GAF file I created earlier to check that we have all the annotations in the GAF file.

**Comment by ValWood** [comment]
I don't know if this indicates some duplication but:

 cat pombase-build-contig-files-only.gaf.txt |wc
   21455  

grep "GOid=GO:" *.contig |wc
   16093  

This could be explained if you expanded the 'piped' extensions into individual annotations.
But definitely not to few!

**Comment by ValWood** [comment]
> I'm now doing a test load of the contigs into a fresh Chado database but with the GO loading disabled.

clever!

**Comment by ValWood** [comment]
It looks good to me!

**Comment by kimrutherford** [comment]
> This could be explained if you expanded the 'piped' extensions into individual annotations.

Yep, that's happening.  The extensions are expanded when loading into Chado.

**Comment by kimrutherford** [comment]
> cat pombase-build-contig-files-only.gaf.txt |wc
> 21455

That count will include ND lines which aren't helpful in this case.

**Comment by mah11** [comment]
> I'm now doing a test load ...

Nice. Assuming it passes your test-load checks, good to ... er, GO ;)

**Comment by kimrutherford** [comment]
> I'm now doing a test load of the contigs into a fresh Chado database but with the GO loading disabled. Then I'll load the GAF file I created earlier to check that we have all the annotations in the GAF file.

That more or less worked.

A couple of annotations didn't end up in the GAF file because they on pseudogenes: SPCC553.05c/wtf6 and SPCC285.06c/wtf17
Now I know that I'll tweak the GAF writer to include them next time I export.

The other issue is that there are some multi-transcript genes where there are annotations on more than one transcript (SPAC27F1.09c/prp10).  The GAF writer ignores transcripts and so in the GAF file it looks like all the annotation is attached to the gene, and is repeated:
```
PomBase    SPAC27F1.09c    prp10           GO:0005634      PMID:10759889   HDA             C       U2 snRNP-associated protein SF3B1       sap155|SF3b155  protein taxon:4896      20050518      PomBase
PomBase    SPAC27F1.09c    prp10           GO:0005634      PMID:10759889   HDA             C       U2 snRNP-associated protein SF3B1       sap155|SF3b155  protein taxon:4896      20050518      PomBase
PomBase    SPAC27F1.09c    prp10           GO:0005634      PMID:10759889   HDA             C       U2 snRNP-associated protein SF3B1       sap155|SF3b155  protein taxon:4896      20050518      PomBase
```

When we read that GAF file all the annotation gets attached to the first transcript (SPAC27F1.09c.1) because the GAF reader doesn't know any better.
There's an open related issue: https://github.com/pombase/pombase-chado/issues/830

Is there preferred way to specify a transcript ID in a GAF file?

A temporary fix is to leave those multi-transcript annotations in the .contig files and move the rest to a GAF fie.

**Comment by mah11** [comment]
> Is there preferred way to specify a transcript ID in a GAF file?

Spec is here: http://geneontology.org/docs/go-annotation-file-gaf-format-2.2/

... but I'm not sure it's very helpful. It's hinting that transcript isoforms can be distinguished using column 17, but it doesn't directly address the case where annotations refer to the protein products of different transcripts from one gene (i.e. our use case). Most of what it has about transcripts seems to be for functional ncRNAs; otherwise it seems to assume we'd have separate protein identifiers for transcript isoform products.

We'll probably have to ask via a ticket or email. Or we could see what happens if we stick one of our transcript isoform IDs in column 17 and pass it off as also an identifier for the protein it encodes.

Tangent: the GYAD/GPI spec is similarly opaque.

> A temporary fix is to leave those multi-transcript annotations in the .contig files and move the rest to a GAF file.

For the small number we're dealing with, that would be OK with me. Life will be easiest if you can give me a list, although in a pinch I could probably remember which genes are affected.

**Comment by ValWood** [comment]
I would go ahead. we can open a ticket about the need to fix the column 17 ones.

Let me know when I can edit... I already have a list of fixes, additions, removals etc  (not log file ones)

**Comment by mah11** [comment]
That all sounds awesome :D

**Comment by kimrutherford** [comment]
> I'm wondering how come there is GO annotation in the contigs associated with multi-gene transcripts. I presumed this would all be from publications and so would be in Canto.

Probably because the annotation can't be attached to a transcript in Canto, only to a gene.  Some of the annotation in the contigs is transcript specific.

**Comment by kimrutherford** [comment]
The load has finished.  There are quite a few things to fix:

  - I forgot to remove the ND lines from `legacy_go_annotations_from_contigs.txt` so they got loaded - whoops
  - the qualifiers column wasn't loaded from `legacy_go_annotations_from_contigs.txt`
  - the column 17 values aren't loaded

I should be able to fix those problems before the next load (on Friday night) so I won't undo all the changes.

I'm about to remove the ND lines and then I'll work on the other problems.

**Comment by kimrutherford** [comment]
Other problems:

  - I forgot to include the annotation from pseudogenes in the GAF file - I'm fixing that now
  - I'm not sure qualifiers like this are being handled correctly: "from=GO:0140053|GO:0005739"
     - but maybe it's better?  (let's chat on the next call):
        - http://dev.pombase.kmr.nz/gene/SPAP8A3.11c
        - https://www.pombase.org/gene/SPAP8A3.11c

**Comment by mah11** [comment]
>  Is it worth fixing `from` entries in the contig files?

answer from meeting: nah, it turned out correct in the GAF so it's all sorted. Yay!

**Comment by ValWood** [comment]
I opened 3 new tickets
https://github.com/pombase/curation/issues/3137
https://github.com/pombase/curation/issues/3136
https://github.com/pombase/canto/issues/2495
to continue when this one closes tomorrow (hopefully)
